### PR TITLE
[IA-4377] add validhosts to relaylistener

### PR DESCRIPTION
**Pull request by n8landolt**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4377

Terra-app PR: https://github.com/DataBiosphere/terra-app/pull/37

### What

-

### Why

-

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 1]
```diff
@@ -488,7 +492,7 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         raw"relaylistener.runtimeName=${appName.value}",
         raw"relaylistener.image=${config.listenerImage}",
         raw"""relaylistener.removeEntityPathFromHttpUrl="${removeEntityPathFromHttpUrl.toString}"""",
-
+        raw"""relaylistener.validHosts=[${validHosts.mkString("','")}]'""",
```
Comment (n8landolt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: Currently unsure the syntax the values file needs for this... searching for examples.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 2]
```diff
@@ -155,7 +156,9 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
             relayPrimaryKey,
             app.appType,
             params.workspaceId,
-            app.appName
+            app.appName,
+            refererConfig.validHosts + params.landingZoneResources.relayNamespace.value
```
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: You can probably just reuse the `relayDomain` variable? (defined on L141)

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala (hunk 1 above)
Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: Maybe just
```
raw"""relaylistener.validHosts=${validHosts.mkString(",")}""",
```
?

So you're not really passing it as an array, but a string to helm. Which is in turn setting the `LISTENER_CORSSUPPORTPROPERTIES_VALIDHOSTS` EV.

Comment (n8landolt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: I'll give it a shot

File: Dockerfile [hunk 3]
```diff
@@ -61,6 +61,10 @@ RUN helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx && \
 # pulling `terra-app-setup` locally and add cert files to the chart.
 # Leonardo will install the chart from local version.
 # We are also caching charts so they are not downloaded with every helm-install
+COPY ./terra-app-setup-0.0.10.tgz /leonardo
```
Comment (rtitle) [inline Dockerfile]: Quick reminder to revert before merging

File: terra-app-setup-0.0.10.tgz
Comment (rtitle) [inline terra-app-setup-0.0.10.tgz:1]: Looks like this was committed.. probably should `git rm` before merging

File: http/src/main/resources/reference.conf [hunk 4]
```diff
@@ -1034,7 +1034,7 @@ terra-app-setup-chart {
   # cert files into the local chart.
   chart-name = "/leonardo/terra-app-setup"
   # If you change this here, be sure to update it in the dockerfile
-  chart-version = "0.0.9"
+  chart-version = "0.0.10"
```
Comment (rtitle) [inline http/src/main/resources/reference.conf]: May need to update our favorite `ConfigReaderSpec`?

**Review by rtitle — APPROVED** [review]

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 5]
```diff
@@ -488,6 +501,10 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         raw"relaylistener.runtimeName=${appName.value}",
         raw"relaylistener.image=${config.listenerImage}",
         raw"""relaylistener.removeEntityPathFromHttpUrl="${removeEntityPathFromHttpUrl.toString}"""",
+        raw"relaylistener.validHosts[0]=${validHosts(0)}",
+        raw"relaylistener.validHosts[1]=${validHosts(1)}",
+        raw"relaylistener.validHosts[2]=${validHosts(2)}",
+        raw"relaylistener.validHosts[3]=${validHosts(3)}",
```
Comment (n8landolt) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: Should this be usable with an unknown number of valid hosts?

Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: does the valid hosts remain the same regardless of environment? If not, yeah I'll have to change this

Comment (lucymcnatt) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala]: just updated it to process a list of any size

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala [hunk 6]
```diff
@@ -469,6 +478,11 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
     // strip the entity path.
     val removeEntityPathFromHttpUrl = appType != AppType.HailBatch
 
+    // validHosts can have a different number of hosts, this pre-processes the list as separate chart values
```
Comment (LizBaldo) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala:481]: Nice 👍

**Review by LizBaldo — APPROVED** [review]

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Comment by lucymcnatt** [comment]
jenkins retest

**Pull request merged**